src/games.c

```diff
@@ -1691,7 +1691,28 @@ int igraph_callaway_traits_game (igraph_t *graph, igraph_integer_t nodes,
     igraph_real_t maxcum;
     igraph_vector_t nodetypes;
 
-    /* TODO: parameter checks */
+    //parameter check 
+    if(nodes < 0){
+        IGRAPH_ERROR("Invalid number of vertices", IGRAPH_EINVAL);
+    }
+    
+    if (types < 1) {
+        IGRAPH_ERROR("types must be >= 1", IGRAPH_EINVAL);
+    }
+
+    if (igraph_vector_size(type_dist) != types) {
+        IGRAPH_ERROR("Number of types is not consistent to type_dist", IGRAPH_EINVAL);
```

Comment:
"The vertex type distribution vector must agree in length with the number of types"

The error concerns `types_dist`, so let's put it first. Again, we spell out what `types_dist` is, for the sake of high-level interfaces.